GitHub_LearningTests/Feature/StartUp/Application/StartUpControllerTests.swift

```diff
@@ -0,0 +1,226 @@
+//
+//  StartUpTests.swift
+//  GitHub_LearningTests
+//
+//  Created by Amin Ashhab on 07.12.2020.
+//  Copyright © 2020 Amin_Second_Test_Project. All rights reserved.
+//
+
+import XCTest
+import CleanCore
+@testable import GitHub_Learning
+
+class StartUpControllerTests: XCTestCase {
+    private var sut: StartUpControllerImpl!
+    private var loadFinishedFacade: LoadOnboardingFinishedFacade!
+    private var keychainController: CredentialKeychainController!
+    private var loginLauncher: LoginLauncherController!
+    private var wf: Wireframe!
+    private var dashboardLauncher: DashboardLauncherController!
+    
+    private func setUpTests(
+        loadOnboardingFinishedFacade: LoadOnboardingFinishedFacade? = nil,
+        credentialKeychainController: CredentialKeychainController? = nil,
+        wireframe: Wireframe? = nil,
+        loginLauncherController: LoginLauncherController? = nil,
+        dashboardLauncherController: DashboardLauncherController? = nil
+    ) {
+        self.loadFinishedFacade = loadOnboardingFinishedFacade ?? LoadOnboardingFinishedFacadeDummy()
+        self.keychainController = credentialKeychainController ?? CredentialKeychainControllerDummy()
+        self.loginLauncher = loginLauncherController ?? LoginLauncherControllerDummy()
+        self.wf = wireframe ?? TestWireframe()
+        self.dashboardLauncher = dashboardLauncherController ?? DashboardLauncherControllerDummy()
+        sut = StartUpControllerImpl(
+            loadOnboardingFinishedFacade: loadFinishedFacade,
+            loginLauncherController: loginLauncher,
+            credentialKeychainController: keychainController,
+            dashboardLauncherController: dashboardLauncher,
+            wireframe: wf)
+    }
+    
+    func test_givenLoadOnboardingFinishedFacadeIsCalledWithSuccessAndResponseIsTrue_whenStartUpIsCalled_thenLoadCredentialsIsCalled() {
```

Comment:
Pro citelnost bych to pojmenoval `test_givenOnboardingIsFinished_whenStartUpIsCalled_thenLoadCredentialsIsCalled `